pkg/virt-controller/watch/vmi.go

```diff
@@ -2131,3 +2149,78 @@ func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachin
 
 	return nil
 }
+
+func (c *VMIController) updateInterfaceStatus(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	originalVMIStatus := vmi.Status.DeepCopy().Interfaces
+	indexedVMIStatus := indexIfaceStatus(originalVMIStatus)
+
+	var newStatus []virtv1.VirtualMachineInstanceNetworkInterface
+	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	for _, network := range vmi.Spec.Networks {
+		status := virtv1.VirtualMachineInstanceNetworkInterface{Name: network.Name}
```

Comment:
If I understand correctly, when a a new iface is defined on the VMI spec, it triggers changing the pod annotation to attach to a network. Up until now, nothing new appears in the interfaces status of the VMI.
Once multus reports that the network is in place (on the pod annotation), we create here a new status entry for the interface in the VMI. And at this stage it has no `InforSource`, nor other information.

If this is correct, then we have a period in which the VMI status reports an interface that is not yet fully plugged. As this is a user facing API, I think we better hint about the source of this data.
In theory, it may remain in this state for some time, especially if the plug fails.

So what do you think about adding a new source? (e.g. `multus-status`).
We do not need to preserve this source, i.e. when it is detected from the domain, we can override it and keep the same behavior as today.

Reply:
1. handler tells launcher it needs to sync.
2. launcher spots interfaces to be hotplugged as a result of this sync
3. launcher hotplugs said interfaces
4. handler gets the result of the sync gRPC from launcher
5. handler computes the updated status, which sees the interface reported in the domain
6. handler finishes its reconcile loop

I see no reason to introduce a dummy state.

Anyway, if you really think this is needed, let's please do that as an optimization in a follow-up PR.

Reply:
You have started from a much later step, where the controller is not in the picture.
There is now a period of time where the virt-controller marks the interface as "configured by CNI" and reports it on the VMI status. Until the virt-handler/virt-launcher completes the hotplug, we have an interface reported with no idea who reported it and why it is there.
We need to express in the API the source of this, so someone who queries the API knows what to do or what to expect.

I do not understand why you call it a dummy state, it is a period in which we report something partial and this period can be non-negligible if we fail to hotplug.

@AlonaKaplan , what do you think?

Reply:
@EdDev in how many situations can a user end up with an interfaces without a defined info source ?

Reply:
Until this change, my understanding that it cannot happen.
The whole logic is [here](https://github.com/kubevirt/kubevirt/blob/1c54a9bd9935055667b188ef51383717e950952e/pkg/network/setup/netstat.go#L85).

Reply:
IMO, since this scenario can happen in exactly one way - an unsuccessful attempt to hotplug an interface - I wouldn't bother with adding a new info source.

Reply:
No one will remember this in 3 months from now and we will need to dig deep into the code to find out.
Also, what is true today may change tomorrow when this will no longer be the case.

Anyway, I can let this go and suggest adding it in a follow up. Please keep it unresolved so I’ll identify which points need to treated.

Reply:
> in how many situations can a user end up with an interfaces without a defined info source ?

new virt-handler - old virt-launcher. If you hot plug to such a VM you will have this situation.